modules/extensions-support/src/main/java/org/mule/module/extension/internal/introspection/describer/AnnotationsBasedDescriber.java

```diff
@@ -297,14 +300,22 @@ private void declareMessageSource(HasSourceDeclarer declarer, Class<? extends So
                 .whichReturns(typeLoader.load(sourceGenerics.get(0)))
                 .withAttributesOfType(typeLoader.load(sourceGenerics.get(1)))
                 .withExceptionEnricherFactory(getExceptionEnricherFactory(sourceType))
-                .withModelProperty(new ImplementingTypeModelProperty(sourceType));
+                .withModelProperty(new ImplementingTypeModelProperty(sourceType))
+                .withMetadataResolverFactory(getMetadataResolverFactory(extensionType, sourceType));
+
+        declareSingleParameters(getParameterFields(sourceType), source, (ModelPropertyContributor) MuleExtensionAnnotationParser::parseMetadataAnnotations);
+        declareParameterGroups(sourceType, source);
 
-        declareAnnotatedParameters(sourceType, source);
```

Comment:
I opened the declaration of source parameters to be more flexible and be able to contribute with ModelProperties at declaration time